hera_cal/lstbin.py

```diff
@@ -0,0 +1,893 @@
+    antenna_names = map(lambda a: "HH{}".format(a), antenna_numbers)
+
+    # get antpos and uvw
+    antenna_positions = np.array(map(lambda k: antpos[k], antenna_numbers))
+    uvw_array = np.array([antpos[k[0]] - antpos[k[1]] for k in zip(ant_1_array, ant_2_array)])
+
+    # get zenith location
+    zenith_dec_degrees = np.ones_like(baseline_array) * dec
+    zenith_ra_degrees = utils.JD2RA(time_array, longitude)
+    zenith_dec = zenith_dec_degrees * np.pi / 180
+    zenith_ra = zenith_ra_degrees * np.pi / 180
+
+    # instantiate object
+    uvd = UVData()
+
+    # assign parameters
+    params = ['Nants_data', 'Nants_telescope', 'Nbls', 'Nblts', 'Nfreqs', 'Npols', 'Nspws', 'Ntimes',
+              'ant_1_array', 'ant_2_array', 'antenna_names', 'antenna_numbers', 'baseline_array',
+              'channel_width', 'data_array', 'flag_array', 'freq_array', 'history', 'instrument',
+              'integration_time', 'lst_array', 'nsample_array', 'object_name', 'phase_type',
+              'polarization_array', 'spw_array', 'telescope_location', 'telescope_name', 'time_array',
+              'uvw_array', 'vis_units', 'antenna_positions', 'zenith_dec', 'zenith_ra']              
+    for p in params:
+        uvd.__setattr__(p, locals()[p])
+
+    # write uvdata
+    if write_miriad:
+        # check output
+        fname = os.path.join(outdir, fname)
+        if os.path.exists(fname) and overwrite is False:
+            abscal.echo("{} exists, not overwriting".format(fname), verbose=verbose)
+        else:
+            abscal.echo("saving {}".format(fname), type=0, verbose=verbose)
+            uvd.write_miriad(fname, clobber=True)
+
+    if return_uvdata:
+        return uvd
+
+
+def sigma_clip(array, sigma=4.0, axis=0):
+    """
+    one-iteration sigma clipping algorithm
+
+    Parameters:
+    -----------
+    array : ndarray of complex visibility data
+
+    sigma : float, sigma threshold to cut above
+
+    axis : int, axis of array to sigma clip
+
+    Output:
+    -------
+    array : same as input array, but clipped values have been set to np.nan
+    """
+    # ensure array is an array
+    array = np.array(array)
+
+    # get robust location
+    mean = np.nanmedian(array, axis=axis)
+
+    # get robust std
+    std = np.sqrt(astats.biweight_midvariance(array, axis=axis))
+
+    # set cut data to nans
+    cut = np.where(np.abs(array-mean)/std > sigma)
+    array[cut] *= np.nan
+
+    return array
+
+
+def unwrap(array, wrap_point=2*np.pi):
+    """
+    Unwrap 1D LST ndarray that is periodic about wrap_point.
+
+    Parameters:
+    -----------
+    array : type=ndarray, 1D array containing LST values in units of wrap_point.
+
+    wrap_point : type=float, full day of LST.
+
+    Output:
+    -------
+    new_array : type=ndarray, 1D array with unwrapped LST values
+    """
+    array = copy.copy(array)
+    new_array = np.empty_like(array)
+    for i, v in enumerate(array):
+        if i == 0:
+            while v < 0:
```

Comment:
This function seems to be able to add wrap_point to array[0] many times but can only add it once to array[1:]. Perhaps you need a check to make sure all data are within a wrap_point of one another.

Also, it's a bit strange to have an if i == 0 in the loop...why not do it outside the loop?

Also, why not use np.unwrap?